src/bors/handlers/pr_events.rs

```diff
@@ -444,4 +459,58 @@ mod tests {
         })
         .await;
     }
+
+    #[sqlx::test]
+    async fn mergeable_queue_processes_pr_base_change(pool: sqlx::PgPool) {
+        run_test(pool, |mut tester| async {
+            tester
+                .edit_pr(default_repo_name(), default_pr_number(), |pr| {
+                    pr.mergeable_state = OctocrabMergeableState::Clean;
+                })
+                .await?;
+            tester
+                .wait_for_default_pr(|pr| pr.mergeable_state == MergeableState::Mergeable)
+                .await?;
+
+            let branch = tester.create_branch("beta").clone();
+            tester
+                .edit_pr(default_repo_name(), default_pr_number(), |pr| {
+                    pr.base_branch = branch;
+                })
+                .await?;
+            tester
```

Comment:
We should wait for the PR having the `Unknown` state here, to make sure that the bot can deal with `Dirty` being set sometime in the future.